powerstore/datasource_protection_policy.go

```diff
@@ -0,0 +1,413 @@
+package powerstore
+
+import (
+	"context"
+	"terraform-provider-powerstore/client"
+	"terraform-provider-powerstore/models"
+
+	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type protectionPolicyDataSource struct {
+	client *client.Client
+}
+
+type protectionPolicyDataSourceModel struct {
+	Policies []models.ProtectionPolicyDataSource `tfsdk:"policies"`
+	ID       types.String                        `tfsdk:"id"`
+	Name     types.String                        `tfsdk:"name"`
+}
+
+var (
+	_ datasource.DataSource              = &protectionPolicyDataSource{}
+	_ datasource.DataSourceWithConfigure = &protectionPolicyDataSource{}
+)
+
+// newProtectionPolicyDataSource returns the protecion policy data source object
+func newProtectionPolicyDataSource() datasource.DataSource {
+	return &protectionPolicyDataSource{}
+}
+
+func (d *protectionPolicyDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_protectionpolicy"
+}
+
+func (d *protectionPolicyDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: ".",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Unique identifier of the policy.",
+				MarkdownDescription: "Unique identifier of the policy.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("name")),
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "Policy name.",
+				MarkdownDescription: "Policy name.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"policies": schema.ListNestedAttribute{
+				Description:         "List of policies.",
+				MarkdownDescription: "List of policies.",
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description:         "Unique identifier of the policy.",
```

Comment:
better to specify protection policy explicitly rather than policy

Reply:
done